Release History &#8212; bedtools 2.31.0 documentation Navigation index next | previous | bedtools v2.31.0 &#187; Release History ¶ Version 2.30.0 (23-Dec-2021) ¶ Thanks to Hao Hou (github: &#64;38), we have substantial improvements in the speed associated with parsing input files and in printing results. It turns our that these tasks consume a large proportion of run time, especially as input files increase in size. These changes result in a 2-3X improvement in speed, depending on input types, options, etc. Thanks to John Marshall (github: &#64;jmarshall), who improved the stability and cleanliness of the code used for random number generation. These changes also squash a bug that arises on Debian systems. John Marshall cleaned up some lingering data type problems in the slop tool. Thanks to &#64;gringer for adding teh -ignoreD option to the genomecov tool, which allows D CIGAR operations to be ignored when calculating coverage. This is useful for long-read technologies with high INDEL error rates. Added a fix for a bug that did not properly handle the splitting of intervals in BED12 records with one block. Thanks to John Marshall (github: &#64;jmarshall), we have addressed numerical instability issues in the fisher tool. Thanks to Hao Hou (github: &#64;38), reference genomes can be read as an environment variable ( CRAM_REFERENCE ) when using CRAM input files. Added a -rna option to the getfasta tool to allow support for RNA genomes. Thanks to Hao Hou (github: &#64;38), we fixed input file format detection bugs arising in ZSH. Thanks to Josh Shapiro (github:&#64;jashapiro) for clarifying a confusing inconcistency in the documentation for the coverage tool. Thanks to Hao Hou (github: &#64;38), we suppressed unnecessary warnings when reading GZIPP’ed files. Thanks to Hao Hou (github: &#64;38), we fixed an overflow bug in the shuffle tool. Thanks to Hao Hou (github: &#64;38), we fixed an data type bug in the shift tool. Thanks to John Marshall (github: &#64;jmarshall) and Hao Hou (github: &#64;38), we have cleaned up the internal support for htslib. Version 2.29.2 (17-Dec-2019) ¶ Fixed a bug that mistakenly removed a BAM/CRAM header line (sorting criteria). Version 2.29.1 (9-Dec-2019) ¶ Fixed a bug that now allows blocked intersection to be counted based on unique base pairs of overlap. The resolution for issue 750 in version 2.29.0 mistakenly allowed for fractional overlap to be counted based upon redundant overlap. Moved to Github Continuous Integration for automatic testing. Fixed a bug that injected erroneous quality values with BAM records had no valid quality values. Fixed a bug that destroyed backwards compatibility in the getfasta tool. Thanks to Torsten Seeman for reporting this. Fixed a corner case bug in the reldist tool. Fixed a bug in the bedtobam tool that caused the last character in read names to be removed. Fixed a bug causing a segfault in the jaccard tool. Fixed a bug causing a corner case issue in the way coordinates are reported in the flank tool. Version 2.29.0 (3-Sep-2019) ¶ Added a new -C option to the intersect tool that separately reports the count of intersections observed for each database ( -b ) file given. Formerly, the -c option reported to sum of all intersections observed across all database files. Fixed an important bug in intersect that prevented some split reads from being counted properly with using the -split option with the -f option. Fixed a bug in shuffle such that shuffled regions should have the same strand as the chose -incl region. Added a new -L option to L`imit the output of the `complement tool to solely the chromosomes that are represented in the -i file. Fixed a regression in the multicov tool introduced in 2.28 that caused incorrect counts. Added support for multi-mapping reads in the bamtofastq tool. Fixed a bug that prevented the “window” tool from properly adding interval “slop” to BAM records. Fixed a bug that caused the slop tool to not truncate an interval’s end coordinate when it overlapped the end of a chromosome. Added support for the “=” and “X” CIGAR operations to bamtobed . Various other minor bug fixes and improvements to documentation. Version 2.28.0 (23-Mar-2019) ¶ Included support for htslib to enable CRAM support and long-term stability (Thanks to Hao Hou!) Included support for genomes with large chromosomes by moving to 64-bit integeres throughout the code base. Thanks to Brent Pedersen and John Marshall! We now provide a statically-linked binary for LINUX (not OSX) systems. Various minor bug fixes. Version 2.27.0 (6-Dec-2017) ¶ Fixed a big memory leak and algorithmic flaw in the split option. Thanks to Neil Kindlon! Resolved compilation errors on OSX High Sierra. Many thanks to &#64;jonchang! Fixed a bug in the shift tool that caused some intervals to exceed the end of the chromosome. Thanks to &#64;wlholtz Fixed major bug in groupby that prevented proper functionality. Speed improvements to the shuffle tool. Bug fixes to the p-value calculation in the fisher tool. Thanks to Brent Pedersen. Allow BED headers to start with chrom or chr Fixes to the “k-closest” functionality in the closest tool. Thanks to Neil Kindlon. Fixes to the output of the freqasc, freqdesc, distinct_sort_num and distinct_sort, and num_desc operations in the groupby tool. Thanks to &#64;ghuls. Many minor bug fixes and compilation improvements from Luke Goodsell. Added the -fullHeader option to the maskfasta tool. Thanks to &#64;ghuls. Many bug fixes and performance improvements from John Marshall. Fixed bug in the -N/-f behavior in subtract . Full support for .fai files as genome ( -g ) files. Many other minor bug fixes and functionality improvements. Version 2.26.0 (7-July-2016) ¶ Fixed a major memory leak when using -sorted . Thanks to Emily Tsang and Stephen Montgomery. Fixed a bug for BED files containing a single record with no newline. Thanks to &#64;jmarshall. Fixed a bug in the contigency table values for thr fisher tool. The getfasta tool includes name, chromosome and position in fasta headers when the -name option is used. Thanks to &#64;rishavray. Fixed a bug that now forces the coverage tool to process every record in the -a file. Fixed a bug preventing proper processing of BED files with consecutive tabs. VCF files containing structural variants now infer SV length from either the SVLEN or END INFO fields. Thanks to Zev Kronenberg. Resolve off by one bugs when intersecting GFF or VCF files with BED files. The shuffle tool now uses roulette wheel sampling to shuffle to -incl regions based upon the size of the interval. Thanks to Zev Kronenberg and Michael Imbeault. Fixed a bug in coverage that prevented correct calculation of depth when using the -split option. The shuffle tool warns when an interval exceeds the maximum chromosome length. The complement tool better checks intervals against the chromosome lengths. Fixes for stddev , min , and max operations. Thanks to &#64;jmarshall. Enabled stdev , sstdev , freqasc , and freqdesc options for groupby . Allow -s and -w to be used in any order for makewindows . Added new -bedOut option to getfasta . The -r option forces the -F value for intersect . Add -pc option to the genomecov tool, allowing coverage to be calculated based upon paired-end fragments. Version 2.25.0 (3-Sept-2015) ¶ Added new -F option that allows one to set the minimum fraction of overlap required for the B interval. This complements the functionality of the -f option.Available for intersect , coverage , map , subtract , and jaccard . Added new -e option that allows one to require that the minimum fraction overlap is achieved in either A _OR_ B, not A _AND_ B which is the behavior of the -r option. Available for intersect , coverage , map , subtract , and jaccard . Fixed a longstanding bug that prevented genomecov from reporting chromosomes that lack a single interval. Modified a src directory called “aux” to “driver” to prevent compilation errors on Windows machines. Thanks very much to John Marshall. Fixed a regression that caused the coverage tool to complain if BED files had less than 5 columns. Fixed a variable overload bug that prevented compilation on Debian machines. Speedups to the groupby tool. New -delim option for the groupby tool. Fixed a bug in map that prevented strand-specifc overlaps from being reported when using certain BEDPLUS formats. Prevented excessive memory usage when not using pre-sorted input. Version 2.24.0 (27-May-2015) ¶ The coverage tool now takes advantage of pre-sorted intervals via the -sorted option. This allows the coverage tool to be much faster, use far less memory, and report coverage for intervals in their original order in the input file. We have changed the behavior of the coverage tool such that it is consistent with the other tools. Specifically, coverage is now computed for the intervals in the A file based on the overlaps with the B file, rather than vice versa. The subtract tool now supports pre-sorted data via the -sorted option and is therefore much faster and scalable. The -nonamecheck option provides greater tolerance for chromosome labeling when using the -sorted option. Support for multiple SVLEN tags in VCF format, and fixed a bug that failed to process SVLEN tags coming at the end of a VCF INFO field. Support for reverse complementing IUPAC codes in the getfasta tool. Provided greater flexibility for “BED+” files, where the first 3 columns are chrom, start, and end, and the remaining columns are free-form. We now detect stale FAI files and recreate an index thanks to a fix from &#64;gtamazian. New feature from Pierre Lindenbaum allowing the sort tool to sort files based on the chromosome order in a faidx file. Eliminated multiple compilation warnings thanks to John Marshall. Fixed bug in handling INS variants in VCF files. Version 2.23.0 (22-Feb-2015) ¶ Added -k option to the closest tool to report the k-closest features in one or more -b files. Added -fd option to the closest tool to for th